File: 3rd Sem/NM/python/linearregression.py
# Input x and y values
x = list(map(float, input("Enter x values (space-separated): ").split()))
y = list(map(float, input("Enter y values (space-separated): ").split()))

# Number of data points
n = len(x)

# Means of x and y
x_mean = sum(x) / n
y_mean = sum(y) / n

# Summations needed for the slope (b1)
xy_sum = sum(x[i] * y[i] for i in range(n))
xx_sum = sum(x[i] ** 2 for i in range(n))

# Calculating coefficients
b1 = (xy_sum - n * x_mean * y_mean) / (xx_sum - n * x_mean ** 2)
b0 = y_mean - b1 * x_mean

# Output the equation
print(f"Linear Regression Equation: y = {b0:.4f} + {b1:.4f}x")

File: 3rd Sem/NM/C-program/linearregression.c
#include <stdio.h>

int main() {
    int n;
    printf("Enter the number of data points: ");
    scanf("%d", &n);

    double x[n], y[n];
    printf("Enter x values (space-separated): ");
    for (int i = 0; i < n; i++) {
        scanf("%lf", &x[i]);
    }

    printf("Enter y values (space-separated): ");
    for (int i = 0; i < n; i++) {
        scanf("%lf", &y[i]);
    }

    double x_mean = 0, y_mean = 0;
    for (int i = 0; i < n; i++) {
        x_mean += x[i];
        y_mean += y[i];
    }

    x_mean /= n;
    y_mean /= n;

    double xy_sum = 0, xx_sum = 0;
    for (int i = 0; i < n; i++) {
        xy_sum += x[i] * y[i];
        xx_sum += x[i] * x[i];
    }

    double b1 = (xy_sum - n * x_mean * y_mean) / (xx_sum - n * x_mean * x_mean);
    double b0 = y_mean - b1 * x_mean;

    printf("Linear Regression Equation: y = %.6lf + %.6lfx\n", b0, b1);

    return 0;
}
